lib/src/app.dart

```diff
@@ -106,6 +112,56 @@ class _AppState extends ConsumerState<Application> {
       setOptimalDisplayMode();
     }
 
+    ref.read(localNotificationServiceProvider).init();
+
+    ref.listenManual(challengesProvider, (prev, current) {
+      final prevIds = prev!.value!.inward.map((challenge) => challenge.id);
+      final inward = current.value!.inward;
+      final repo = ref.read(challengeRepositoryProvider);
+      final l10n = ref.read(l10nProvider).strings;
+
+      inward
+          .map((element) => element.id)
+          .where((id) => !prevIds.contains(id))
```

Comment:
I wonder if this check is enough to prevent sending twice the notification for a same challenge. I assume there are other checks, like ensuring the notification ID is linked to a specific challenge to not send it twice?

Reply:
That id is from the server, i would assume that lichess ensures uniqueness .